### DISCUSSION TITLE
Top level jupyter toolbar feels alien in VS Code UI

### DISCUSSION THREAD

**Discussion by isidorn** [microsoft/vscode-jupyter#9147] [New Ideas]
Top level Jupyter toolbar feels alien in the VS Code UI imho. We do not use editor level toolbars in other places.
Why are some of the actions needed here? For example "Outline" and "Variables". Outline is nothing Notebook specific.

![Screenshot 2022-02-28 at 18 01 22](https://user-images.githubusercontent.com/1926584/156025813-1f8a697f-c79b-4a6c-a249-6f65b6e6a331.png)

fyi @minsa110

**Comment by greazer** [comment]
Thanks for the feedback @isidorn.

Based on our usability feedback from many users, not having a per-notebook toolbar was confusing in the least, preventing usage at worst. Every other notebook implementation in the wild that we are aware of (and there are many) has a similarly located and populated toolbar.

We have had countless requests for "Table of Contents" support for notebooks, even though it's been there for a long time via "Outline". To date we still receive requests for TOC even though this button is here. 

Variables is indeed specific to notebooks. In fact, it's specific per notebook (actually kernel), and arguably should be part of the notebook itself (not a global VS Code window).

**Reply by lydavidzhang** [reply]
I love vscode UI but about the TOC or Outline whatever, I can't find the detailed settings to solve my problems:

I use jupyter notebook for learning data science,  the "Outline" not only list the markdown titles but also the whole contents. It makes me very difficult to find the exactly section I need. I've found a setting about code cell, I hope there is a setting for the markdown contents too. Or just list the titles, I think it's maybe enough.

Also if it can automatically generate the title serial number will be very helpful. 

I don't know how to add the TOC at the top my notebook, if it can be done by [TOC] or "Outline" button will be very helpful. So that I can easily export my notebook into PDF or DOC with it.

**Comment by greazer** [comment]
I should mention that turning off this setting will remove that toolbar from the notebook, as the "alien-ness" was something that was discussed quite a bit when notebooks were first implemented.

<img width="734" alt="image" src="https://user-images.githubusercontent.com/11685408/156028387-390444fc-41d3-4494-8384-3b5fbf577ec4.png">

**Comment by isidorn** [comment]
Thank you for letting me know. I was aware that a big part of the UI is customisable, I just have strong feelings towards what are the defaults.

My biggest concern with the current UI is that it just does not work on smaller screens. I suggest that the Jupyter team tries self-hosting for a day on 15-inch laptops to get a feeling of what I am trying to convey.

**Reply by greazer** [reply]
That's actually really good feedback. It isn't necessarily about the fact that the toolbar seems alien to VS Code, it's more that it makes using notebooks tougher on smaller screens.
